Server/Quality_of_Life_changer.WebApi/Controllers/UserProfileController.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace Quality_of_Life_changer.WebApi.Controllers;
+
+using Contracts.Commands;
+using FluentValidation;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using Model.UserProfile;
+using System.Text;
+using ValidationException = Contracts.Exceptions.ValidationException;
+
+[Route("api/user/{userId}/profile")]
+[ApiController]
+public class UserProfileController : ControllerBase
+{
+    private readonly IValidator<CalendarModel> _calendarModelValidator;
+    private readonly IMediator _mediator;
+
+    public UserProfileController(IMediator mediator, IValidator<CalendarModel> calendarModelValidator)
+    {
+        _mediator = mediator;
+        _calendarModelValidator = calendarModelValidator;
+    }
+
+    [HttpPost("calendars")]
+    public async Task<IActionResult> TodayEvents([FromBody] CalendarModel model, string userId)
+    {
+        var result = await _calendarModelValidator.ValidateAsync(model);
+
+        if (!result.IsValid)
+        {
+            var stringBuilder = new StringBuilder();
+            foreach (var error in result.Errors)
+            {
+                stringBuilder.Append(error.ErrorMessage);
+            }
+
+            throw new ValidationException(stringBuilder.ToString());
```

Comment:
we don't need an exception here, because it's not a server-side issue. so no reason to log it.